classes/blocks/class-enform.php

```diff
@@ -2,158 +2,144 @@
 /**
  * ENForm block class
  *
- * @package P4GEN
+ * @package P4GBKS
  */
 
-namespace P4GEN\Blocks;
+namespace P4GBKS\Blocks;
 
-use P4GEN\Views\View;
-use P4GEN\Controllers\Ensapi_Controller as Ensapi;
+use P4GBKS\Views\View;
+use P4GBKS\Controllers\Ensapi_Controller as Ensapi;
+use WP_Block_Type_Registry;
 
 /**
  * Class ENForm
  *
- * @package P4GEN\Blocks
+ * @package P4GBKS\Blocks
  */
 class ENForm extends Base_Block {
 
 	/** @const string BLOCK_NAME */
-	const BLOCK_NAME = 'enform';
+	protected const BLOCK_NAME = 'enform';
 
 	/**
 	 * Page types for EN forms
 	 *
 	 * @const array ENFORM_PAGE_TYPES
 	 */
-	const ENFORM_PAGE_TYPES = [ 'PET', 'EMS' ];
-
-	/**
-	 * ENSAPI Object
-	 *
-	 * @var Ensapi $ensapi
-	 */
-	private $ens_api = null;
+	public const ENFORM_PAGE_TYPES = [ 'PET', 'EMS' ];
 
 	/**
 	 * Custom meta field where fields configuration is saved to.
 	 */
-	const FIELDS_META = 'p4enform_fields';
-
-	/**
-	 * Class Loader reference
-	 *
-	 * @var \P4GEN\Loader $loader;
-	 */
-	private $loader;
+	private const FIELDS_META = 'p4enform_fields';
 
 	/**
 	 * ENForm constructor.
-	 *
-	 * @param array $loader Loader instance.
 	 */
-	public function __construct( $loader ) {
-		$this->loader = $loader;
+	public function __construct() {
 
 		add_shortcode( 'shortcake_enblock', [ $this, 'add_block_shortcode' ] );
 
-		register_block_type(
-			'planet4-blocks/enform',
-			[
-				'render_callback' => [ $this, 'render' ],
-				'attributes'      => [
-					'en_page_id'                    => [
-						'type'    => 'integer',
-						'default' => 0,
-					],
-					'enform_goal'                   => [
-						'type'    => 'string',
-						'default' => '',
-					],
-					'en_form_style'                 => [
-						'type'    => 'string',
-						'default' => 'fullwidth',
-					],
-					'enform_style'                  => [
-						'type'    => 'string',
-						'default' => '',
-					],
-					'title'                         => [
-						'type'    => 'string',
-						'default' => '',
-					],
-					'description'                   => [
-						'type'    => 'string',
-						'default' => '',
-					],
-					'campaign_logo'                 => [
-						'type' => 'boolean',
+		if ( ! WP_Block_Type_Registry::get_instance()->is_registered( 'planet4-blocks/enform' ) ) {
```

Comment:
If we don't need [these two lines](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/308/files#diff-bfb6ea67fe59d361747a005ff3477325R136-R137) to be executed, we could instead do
```
if ( WP_Block_Type_Registry::get_instance()->is_registered( 'planet4-blocks/enform' ) ) {
    return;
}
```
and it wouldn't modify the history of the block definition.